extensions/amp-autocomplete/0.1/amp-autocomplete.js

```diff
@@ -14,43 +14,415 @@
+     * @private {?Element}
+     */
+    this.activeElement_ = null;
+
+    /**
+     * The reference to the <div> that contains template-rendered children.
+     * @private {?Element}
+     */
     this.container_ = null;
   }
 
   /** @override */
   buildCallback() {
     userAssert(isExperimentOn(this.win, 'amp-autocomplete'),
         `Experiment ${EXPERIMENT} is not turned on.`);
-    this.container_ = this.element.ownerDocument.createElement('div');
-    this.container_.textContent = this.myText_;
-    this.element.appendChild(this.container_);
-    this.applyFillContent(this.container_, /* replacedContent */ true);
+
+    this.inlineData_ = this.getInlineData_();
+
+    const inputElements = childElementsByTag(this.element, 'INPUT');
+    userAssert(inputElements.length === 1,
+        `${TAG} should contain exactly one <input> child`);
+    this.inputElement_ = inputElements[0];
+
+    this.filter_ = userAssert(this.element.getAttribute('filter'),
+        `${TAG} requires "filter" attribute.`);
+    userAssert(isEnumValue(FilterType, this.filter_),
+        `Unexpected filter: ${this.filter_}`);
+
+    this.minChars_ = this.element.hasAttribute('min-characters') ?
+      parseInt(this.element.getAttribute('min-characters'), 10) : 1;
+    this.maxEntries_ = parseInt(this.element.getAttribute('max-entries'), 10);
+
+    return this.mutateElement(() => {
+      this.container_ = this.createContainer_();
+      this.element.appendChild(this.container_);
+    });
+  }
+
+  /**
+   * Reads the 'items' data from the child <script> element.
+   * For use with static local data.
+   * @return {?Array}
+   * @private
+   */
+  getInlineData_() {
+    const scriptElements = childElementsByTag(this.element, 'SCRIPT');
+    if (!scriptElements.length) {
+      return null;
+    }
+    userAssert(scriptElements.length === 1,
+        `${TAG} should contain at most one <script> child`);
+    const scriptElement = scriptElements[0];
+    userAssert(isJsonScriptTag(scriptElement),
+        `${TAG} should be inside a <script> tag with type="application/json"`);
+    const json = tryParseJson(scriptElement.textContent,
+        error => {
+          throw error;
+        });
+    return json['items'] ? json['items'] : [];
+  }
+
+  /**
+   * Creates and returns <div> that contains the template-rendered children.
+   * @return {Element}
+   * @private
+   */
+  createContainer_() {
+    const container = this.element.ownerDocument.createElement('div');
+    container.classList.add('i-amphtml-autocomplete-results');
+    container.setAttribute('role', 'list');
+    toggle(container, false);
+    this.applyFillContent(container, /* replacedContent */ true);
+    return container;
   }
 
   /** @override */
   layoutCallback() {
-    // Actually load your resource or render more expensive resources.
-    return Promise.resolve();
+    // Disable autofill in browsers.
+    this.inputElement_.setAttribute('autocomplete', 'off');
```

Comment:
This may not work in all cases: See https://bugs.chromium.org/p/chromium/issues/detail?id=468153#c164 for a recommended solution

Reply:
I followed the recommended solution and it appears to restore Chrome's autofill, whereas `autocomplete=off` disabled it successfully. This is just from manually testing on Chrome and Firefox. Though I can't find more recently published information for when/if Chrome stopped ignoring this attribute, MDN suggests that `autocomplete=off` should work for Firefox, Chrome, and IE: https://developer.mozilla.org/en-US/docs/Web/Security/Securing_your_site/Turning_off_form_autocompletion

Reply:
Ah, it looks like Chrome did make a change to conditionally respect the attribute at the end of 2015: https://bugs.chromium.org/p/chromium/issues/detail?id=468153#c123

An important distinction is that they still ignore the attribute for login fields, but that is OK for our case because autocomplete isn't scoped to support those input types.